NeuroID/src/main/java/com/neuroid/tracker/service/NIDCallActivityListener.kt

```diff
@@ -0,0 +1,152 @@
+package com.neuroid.tracker.service
+
+import android.Manifest
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.content.IntentFilter
+import android.content.pm.PackageManager
+import android.os.Build
+import android.telephony.PhoneStateListener
+import android.telephony.TelephonyCallback
+import android.telephony.TelephonyManager
+import androidx.annotation.RequiresApi
+import androidx.core.app.ActivityCompat
+import com.neuroid.tracker.events.CALL_IN_PROGRESS
+import com.neuroid.tracker.events.CallInProgress
+import com.neuroid.tracker.models.NIDEventModel
+import com.neuroid.tracker.storage.NIDDataStoreManager
+import com.neuroid.tracker.utils.NIDLog
+import com.neuroid.tracker.utils.VersionChecker
+import java.util.Calendar
+
+class NIDCallActivityListener(
+    private val dataStoreManager: NIDDataStoreManager, private val versionChecker: VersionChecker
+) : BroadcastReceiver() {
+    private lateinit var intentFilter: IntentFilter
+    lateinit var intent: Intent
+    private val isReceiverRegistered = false
+
+    @RequiresApi(Build.VERSION_CODES.S)
+    override fun onReceive(context: Context?, intent: Intent?) {
+        if (context != null) {
+            registerCustomTelephonyCallback(context)
+        }
+    }
+
+    internal fun setCallActivityListener(context: Context) {
+        if (ActivityCompat.checkSelfPermission(
+                context, Manifest.permission.READ_PHONE_STATE
+            ) == PackageManager.PERMISSION_GRANTED && !isReceiverRegistered
+        ) {
+            NIDLog.d(msg = "Initializing call activity listener")
+            intentFilter = IntentFilter("android.intent.action.PHONE_STATE")
+            context.registerReceiver(this, intentFilter)
+        } else {
+            NIDLog.d(msg = "Permission to listen to call status not found")
+            saveCallInProgressEvent(9)
+        }
+    }
+
+    fun unregisterCallActivityListener(context: Context?) {
+        if (isReceiverRegistered) {
+            context?.unregisterReceiver(this)
+        }
+    }
+
+    fun saveCallInProgressEvent(state: Int) {
+        when (state) {
```

Comment:
Can we use an enumeration or other means of documenting what each state value means?

Reply:
Good suggestion. Updated Enum and added link to Android doc.